### ISSUE TITLE
[Bug][svelte/indent] invalid indentation in some specific scenarios

### ISSUE THREAD

**Issue by jamesst20** [sveltejs/eslint-plugin-svelte#783]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.4.0

### What version of `eslint-plugin-svelte` are you using?

5.1.3

### What did you do?

<details>
<summary>Configuration</summary>

```
import tsParser from "@typescript-eslint/parser";
import "eslint-plugin-only-warn";
import eslintPluginSvelte from "eslint-plugin-svelte";
import globals from "globals";
import svelteParser from "svelte-eslint-parser";

export default [
  {
    languageOptions: {
      globals: globals.browser,
      parser: tsParser,
    },
  },
  {
    files: ["**/*.svelte"],
    languageOptions: {
      ecmaVersion: 2022,
      parser: svelteParser,
      parserOptions: {
        extraFileExtensions: [".svelte"],
        parser: tsParser,
      },
      sourceType: "module",
    },
  },
  ...eslintPluginSvelte.configs["flat/recommended"],
  {
    files: ["**/*.svelte"],
    rules: {
      indent: "off",
      "svelte/indent": [
        "error",
        {
          alignAttributesVertically: false,
          ignoredNodes: [],
          indent: 2,
          switchCase: 1,
        },
      ],
    },
  },
  {
    ignores: [
      "postcss.config.cjs",
      "tailwind.config.js",
      "app/frontend/rails/",
      "node_modules/",
      "build/",
      ".svelte-kit/",
      "package/",
      "public/rcf-assets-dev/",
      "public/rcf-assets/",
      "tmp/",
      "coverage/",
    ],
  },
];

```
</details>

```svelte
<script context="module" lang="ts">
  type GenericTypes = null | number | string | undefined;
  type GenericKeyType<Opt> = keyof Opt | never | null | undefined;
  type ValueType<Opt, VKey extends GenericKeyType<Opt>, VObj extends boolean> = Opt extends GenericTypes
    ? Opt
    : VKey extends keyof Opt
      ? VObj extends true
        ? Opt
        : Opt[VKey]
      : never;

  const dragDrop = (e: any) => {
    e.preventDefault();
    e.stopPropagation();
    /*
     * Warning: This event is called by the element the item was drop on top of.
     * Since we need to update the position of the dragged item (the source), we must use a
     * global context and the onEnd event.
     */
  };
</script>
```

### What did you expect to happen?

Nothing the code is perfectly valid.

### What actually happened?

![Capture d’écran, le 2024-06-13 à 17 18 16](https://github.com/sveltejs/eslint-plugin-svelte/assets/490234/6d9465c1-70da-4961-80ff-c0a01f758983)
![Capture d’écran, le 2024-06-13 à 17 18 35](https://github.com/sveltejs/eslint-plugin-svelte/assets/490234/a12bf3c0-0b17-437f-a882-f8d3076b6d61)

### Link to **GitHub Repo** with Minimal Reproducible Example

https://sveltejs.github.io/eslint-plugin-svelte/playground/#eJxtksFOwzAMhl/FymlDE5M4lm1cQBw4gASCA+OQNl4JpE6VOEA19u44La00xKl2nP/z78Z7VXmDqlCrWAXbMlSeGL94vVWNN8nhVoHTVEvOcas2WwLgrkW4RsJgqweJI6yBknPwLZ+mxCBB5GCpliCRwZ0lNOd/pTfYZfXqtuWNEN6x8zuQJGPwo6f8Uv9hPGqXcJQv4FFgILaRTPyPLzduy7fpRum9Q025bW74R9jPlDsBXOT6EBbHTSa7QzVfPWrBIeFYOuIMLEmfM+9lPCyGoWXCfCKvEBlM0PVl8K3YnGEBmro5rDewHzR42gaREF/iTifHs/n5WIjs2zsR6lqz9TRVlie/7U7gSQeSFyrg4dVG6DkgQaWdQwNlB/yKgA6bXMixZWzgU0dxJY48AefP7nQi3luqED5RBhECe0it0Yy9uPXRZici6PM8WS23eugsn0SfQoXzRQY0SYZPEUFP8Nr5UrtxO+VXmJ7j6Uqi3v1kZJmDg0y8Wg47vVELFWSVoyr2Kn6gY1xaWSliVZwdDj9Yw/9m

**Comment by ota-meshi** [comment]
I don't think the indentation of the ternary operator is a bug, but the rest of it might be.

**Comment by DollarAkshay** [comment]
I had the same problem today with a switch case. 
It expected 16, I had 12.
After adding 4 more spaces, it said expecting 12 but found 16.


Adding `"svelte/indent": [ `error`, { "indent": 4, "switchCase": 0 }]` and setting the `switchCase` to `0` helped me fix this annoying issue.

**Comment by baseballyama** [comment]
I think this rule can be handled by Prettier, so it doesn’t need to be supported by ESLint.
I’d like to deprecate this rule in v3. What do you think? @ota-meshi

**Comment by ota-meshi** [comment]
In my opinion, support for stylistic rules is still needed.
Prettier is opinionated, so it might not be good for some projects.
ESLint is not a formatter, but the ESLint rule options are flexible, so I think the stylistic rules implemented in the ESLint plugin are still a good option for some projects.

**Comment by baseballyama** [comment]
We discussed this matter with the Svelte core team. Our general stance as maintainers is that we do not actively focus on fixing bugs or adding new features for formatting rules, but we are open to community contributions via PRs. Of course, if @ota-meshi feels it is necessary and wants to submit a PR, that’s perfectly fine. However, to ensure effective maintenance with our limited resources, we would like to lower its relative priority.

**Comment by ota-meshi** [comment]
>  we would like to lower its relative priority.

Yeah, I agree that fixing/adding stylistic rules is a low priority (unless someone in the community proactively works on it.)
